src/network/networkd-dhcp-server.c

```diff
@@ -464,3 +479,180 @@ int config_parse_dhcp_server_sip(
+                return 0;
+        }
+        if (u < 1 || u >= 255) {
+                log_syntax(unit, LOG_ERR, filename, line, 0,
+                           "Invalid DHCP server send raw option, valid range is 1-254, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
+        s = strchr(m, ':');
+        if (!s) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Invalid DHCP server send raw option, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
+        *s++ = '\0';
+        r = dhcp_raw_option_data_type_from_string(m);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Invalid DHCP server send data type, ignoring assignment: %s", p);
+                return 0;
+        }
+
+        type = r;
+        switch(type) {
+        case DHCP_RAW_OPTION_DATA_UINT8:{
+                r = safe_atou8(s, &uint8_data);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse DHCPv4 vendor specific uint8 data, ignoring assignment: %s", s);
+                        return 0;
+                }
+
+                udata = &uint8_data;
+                sz = sizeof(uint8_t);
+                break;
+        }
+        case DHCP_RAW_OPTION_DATA_UINT16:{
+                r = safe_atou16(s, &uint16_data);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse DHCPv4 vendor specific uint16 data, ignoring assignment: %s", s);
+                        return 0;
+                }
+
+                udata = &uint16_data;
+                sz = sizeof(uint16_t);
+                break;
+        }
+        case DHCP_RAW_OPTION_DATA_UINT32: {
+                r = safe_atou32(s, &uint32_data);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse DHCPv4 vendor specific uint32 data, ignoring assignment: %s", s);
+                        return 0;
+                }
+
+                udata = &uint32_data;
+                sz = sizeof(uint32_t);
+
+                break;
+        }
+        case DHCP_RAW_OPTION_DATA_IPV4ADDRESS: {
+                r = in_addr_from_string(AF_INET, s, &addr);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse DHCPv4 vendor specific ipv4address data, ignoring assignment: %s", s);
+                        return 0;
+                }
+
+                sz = sizeof(addr.in.s_addr);
+                break;
+        }
+        case DHCP_RAW_OPTION_DATA_STRING:
+                r = unbase64mem(s, (size_t) -1, &q, &sz);
```

Comment:
Is it necessary that the string is encoded?

Reply:
hmm dropped.